service/worker/deletenamespace/reclaimresources/workflow.go

```diff
@@ -135,11 +135,58 @@ func ReclaimResourcesWorkflow(ctx workflow.Context, params ReclaimResourcesParam
 
 	ctx = workflow.WithTaskQueue(ctx, primitives.DeleteNamespaceActivityTQ)
 
+	namespaceDeleteDelay := params.NamespaceDeleteDelay
+	var cancelDeleteDelay workflow.CancelFunc
+	err := workflow.SetUpdateHandlerWithOptions(ctx, "update_namespace_delete_delay", func(ctx workflow.Context, newNamespaceDeleteDelayStr string) (string, error) {
```

Comment:
this is making the code very hard to read and understand.
Specifically closures on namespaceDeleteDelay and cancelDeleteDelay

I will recommend to do the same this that is done in for example DeploymentSeriesWorkflowRunner.
Create ReclameResourceRunner struct, make namespaceDeleteDelay and cancelDeleteDelay members of that struct.

Extract UpdateNamespaceDeleteDelay update handle into a separate function
Same for update validator.

then it will look like

err := workflow.SetUpdateHandlerWithOptions(ctx, "update_namespace_delete_delay", r.UpdateNamespaceDeleteDelay, 
workflow.UpdateHandlerOptions{
   Validator: r.UpdateHandlerOptionsValidator
}

Reply:
We discussed it offline. I personally believe that closures are and inlined functions provide better readability and will leave it as is.